### PR TITLE
[Maintenance] semi-anual "monthly" go snowflake rebase

### PR DESCRIPTION
**Pull request by madisonchamberlain**
### Description

Rebase last few months. The runtime for the test actually decreased quite a bit! 
Screenshot 2023-05-02 at 4 51 06 PM
See the linked workbook to zoom in, but yellow is the unrebased version, blue is the rebased version 

Decisions I made
* Chunk_downloader.go to not use any of sigmas/Qings modifications after discussing with qlc team
* arrow_chunk.go to not use any of sigmas/Qings modifications after discussing with qlc team
* converter_test.go `TestGoTypeToSnowflake` test to run the same test ran by gosnowflake with `tmode: null` to match gregs changes

### My ask for input tables
* Manually go over `converter.go`, `bind_uploader.go`, and `datatype.go` to determine what changes need to be in there for input tables while still letting in the needed bug fixes and optimizations implemented by snowflake 

### Checklist
- [X] Code compiles correctly
- [X] Run ``make fmt`` to fix inconsistent formats
- [X] Run ``make lint`` to get lint errors and fix all of them
- [X] Modified tests which fail without the change (if possible)
- [ ] All tests passing
- [X] Added rebase data to [the workbook](https://staging.sigmacomputing.io/sigma-on-sigma/workbook/Snowflake-Go-Driver-Updates-Hm0VmWLJZbgi9m4hWayjK/edit?:nodeId=KlTusTP6zP) 
- [X] Extended the README / documentation, if necessary

### CODE REVIEW

File: converter.go [hunk 1]
```diff
@@ -134,21 +151,21 @@ func valueToString(v driver.Value, tsmode snowflakeType) (*string, error) {
...
-		if tm, ok := v.(time.Time); ok {
-			switch tsmode {
-			case dateType:
+		if tm, ok := v.(time.Time); ok && dataType != nil {
+			switch {
+			case dataType.Equals(DataTypeDate):
 				_, offset := tm.Zone()
 				tm = tm.Add(time.Second * time.Duration(offset))
-				s := fmt.Sprintf("%d", tm.Unix()*1000)
+				s := strconv.FormatInt(tm.Unix()*1000, 10)
```
Comment (GregOwen) [inline converter.go:159]: Weird that they changed this `fmt.Sprintf("%d")` to `strconv.FormatInt` but not any of the other ones below...

File: bind_uploader.go [hunk 2]
```diff
@@ -292,12 +292,8 @@ func supportedArrayBind(nv *driver.NamedValue) bool {
 		// TODO SNOW-176486 variant, object, array
 
 		// Support for bulk array binding insertion using []interface{}
-		nvValue := reflect.ValueOf(nv)
-		if nvValue.Kind() == reflect.Ptr {
-			value := reflect.Indirect(reflect.ValueOf(nv.Value))
-			if value.Kind() == reflect.Slice || value.Kind() == reflect.Struct {
-				return true
-			}
+		if isInterfaceArrayBinding(nv.Value) {
```
Comment (GregOwen) [inline bind_uploader.go:295]: I don't believe we've made any changes to this method; seems like the merge conflict detector is operating on the file level?

**Review by GregOwen — APPROVED** [review]
Those three files all look like they're in a good place

File: bind_uploader.go (hunk 2 above)
Comment (madisonchamberlain) [reply bind_uploader.go:295]: I think that the way we have been rebasing forces us to rebase their changes onto their changes sometimes which is very confusing. I had to manually go through and handpick our changes onto their current up-to-date version of this file

File: converter.go (hunk 1 above)
Comment (madisonchamberlain) [reply converter.go:159]: Yeah... there are definitely some interesting changes on their end

**Pull request merged**